src/lib.rs

```diff
@@ -40,27 +54,69 @@ impl Router {
             .map(|val| val.to_str().unwrap_or(""))
             .unwrap_or("");
         match uri_path {
-            "/healthz" => Ok(Response::new(Body::from("OK"))),
+            "/healthz" => Box::new(async { Ok(Response::new(Body::from("OK"))) }.boxed()),
+            "/_reload" => {
+                let res = self.reload();
+                Box::new(
+                    async move {
+                        match res {
+                            Ok(_) => Ok(Response::new(Body::from("OK"))),
+                            Err(e) => Ok(internal_error(&e.to_string())),
+                        }
+                    }
+                    .boxed(),
+                )
+            }
             _ => match self
                 .module_config
+                .read()
+                .unwrap()
                 .handler_for_host_path(host.to_lowercase().as_str(), uri_path)
             {
-                Ok(h) => Ok(h
-                    .module
-                    .execute(
-                        h.entrypoint.as_str(),
-                        req,
-                        client_addr,
-                        self.cache_config_path.clone(),
+                Ok(h) => {
+                    let cache_config_path = self.cache_config_path.clone();
+                    Box::new(
+                        async move {
+                            Ok::<_, hyper::Error>(
+                                h.module
+                                    .execute(
+                                        h.entrypoint.as_str(),
+                                        req,
+                                        client_addr,
+                                        cache_config_path,
+                                    )
+                                    .await,
+                            )
+                        }
+                        .boxed(),
                     )
-                    .await),
+                }
                 Err(e) => {
                     log::error!("error: {}", e);
-                    Ok(not_found())
+                    Box::new(async { Ok(not_found()) }.boxed())
                 }
             },
         }
     }
+
+    /// Reload the config and modules.
+    ///
+    /// This rebuilds the modules list from config, and then re-initializes all
+    /// modules. It will call the `_routes()` method on each module. If caching is
+    /// enabled, it will also clear and recreate the module cache.
```

Comment:
Does this actually clear and recreate the cache?

Reply:
Uh-oh... you may be right. I may have to explicitly clear the cache

Reply:
Okay, so I think that `build_registry()` will refresh the cached items. I am just not sure if it will delete obsolete ones. I am going to merge this PR and then try to rewrite build_registry() in a separate PR